### PR TITLE
Normative: Throw Range in PrepareTemporalFields for +∞ or -∞

### PR DESCRIPTION
**Pull request by FrankYFTang**
Fix one of the issue in https://github.com/tc39/proposal-temporal/issues/1467

@ptomato @justingrant @Ms2ger

### CODE REVIEW

File: spec/abstractops.html [hunk 1]
```diff
@@ -1446,6 +1446,7 @@ <h1>PrepareTemporalFields ( _fields_, _fieldNames_, _requiredFields_ )</h1>
           1. If _property_ is in the Property column of <emu-xref href="#table-temporal-field-requirements"></emu-xref> and there is a Conversion value in the same row, then
             1. Let _Conversion_ represent the abstract operation named by the Conversion value of the same row.
             1. Set _value_ to ? _Conversion_(_value_).
+            1. If _value_ is +∞ or -∞, throw a *RangeError* exception.
```
Comment (ljharb) [inline spec/abstractops.html:1449]: would an alternative approach be to make an AO in the Conversion table that throws when it results in an infinity, and then this line wouldn't be needed?

Comment (FrankYFTang) [reply spec/abstractops.html]: Good Idea! I add a new AO ToInteger and rename ToPositiveIntegerOrInfinite to ToPositiveInteger and change all the places which call them and throw on +/- Infinity. I didn't touch anywhere else which does not throw after calling them.

Comment (FrankYFTang) [reply spec/abstractops.html]: PTAL

**Comment by FrankYFTang** [comment]
This PR has normative change to PrepareTemporalFields and PreparePartialTemporalFields but also a lot of editorial changes on other part of the spec which simplfy the spec by using the new AO (ToInteger)

**Comment by FrankYFTang** [comment]
or should we rename the new AO ToIntegerThrowOnInfinity instead of ToInteger so it won't got confused with the old ToInteger in ECMA262 before?

**Comment by ljharb** [comment]
Renaming sounds like a great idea.

**Comment by FrankYFTang** [comment]
> Renaming sounds like a great idea.

ok, let me take some time to change it later today

**Comment by Ms2ger** [comment]
Fwiw, I used "ToFiniteInteger" in the polyfill.

**Comment by Ms2ger** [comment]
I would also recommend adding the new abstract operation and updating the existing call sites that already throw, in a separate PR. That way we can land those changes before the normative change here.

**Comment by FrankYFTang** [comment]
>Fwiw, I used "ToFiniteInteger" in the polyfill.

I don't like that name becaue it imply Integer could be Infinity while it is not.

> I would also recommend adding the new abstract operation and updating the existing call sites that already throw, in a separate PR. That way we can land those changes before the normative change here.

good point. I like that. I will change that.

**Comment by ptomato** [comment]
@FrankYFTang Have you had time to split out the non-normative part from the normative part? If not, let me know and I can do it.

**Comment by FrankYFTang** [comment]
> @FrankYFTang Have you had time to split out the non-normative part from the normative part? If not, let me know and I can do it.

Sorry, I took the whole week off last week for a personal issue and didn't read my email till now.  I can work on it coming week if it is not urgent.

**Comment by ptomato** [comment]
Hi Frank, that's no problem. I hope everything is OK.

I'd like to get this done in plenty of time to put the normative part on the agenda for next TC39 meeting. If you can do it sometime this week, great. If not, it's no problem - just let me know so I can do it.

**Comment by ptomato** [comment]
I split out the non-normative part into #1711.

**Comment by ptomato** [comment]
I've added tests to this and based it on top of the editorial change in #1711. It should be ready to present for consensus now.

**Comment by ptomato** [comment]
This change achieved consensus in TC39.

**Review by ptomato — APPROVED** [review]
:+1:

**Pull request merged**